File: backend/backend/src/main/java/com/hostelhub/backend/service/AdminService.java
package com.hostelhub.backend.service;
import java.util.List;
import com.hostelhub.backend.dto.response.HostelResponseDto;
import com.hostelhub.backend.dto.admin.AdminStatsResponse;

public interface AdminService {

    AdminStatsResponse getDashboardStats();

    void approveHostel(Long hostelId);

    void rejectHostel(Long hostelId);
    List<HostelResponseDto> getPendingHostels();
}

File: frontend/src/services/adminService.js
import api from "./api";


export const getPendingHostels = async () => {
  const response = await api.get("/admin/hostels/pending");
  return response.data;
};

export const approveHostel = async (id) => {
  const response = await api.put(
    `/admin/hostels/${id}/approve`
  );

  return response.data;
};

export const rejectHostel = async (id) => {
  const response = await api.put(
    `/admin/hostels/${id}/reject`
  );

  return response.data;
};